src/grpc/mod.rs

```diff
@@ -99,60 +125,79 @@ impl KeyProviderService for KeyProvider {
     ) -> Result<Response<KeyProviderKeyWrapProtocolOutput>, Status> {
         debug!("The WrapKey API is called...");
         debug!("WrapKey API is unimplemented!");
-        Err(Status::unimplemented(
-            "WrapKey API of attestation-agent is unimplemented!",
-        ))
+        Err(Status::unimplemented(format!(
+            "WrapKey API of {} is unimplemented!",
+            AGENT_NAME,
+        )))
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
 struct InputPayload {
     kbc_name: String,
     kbs_uri: String,
     annotation: String,
 }
 
-fn parse_input(input_byte: Vec<u8>) -> Result<InputPayload> {
-    let input_string = String::from_utf8(input_byte)?;
-    debug!("UnWrapKey API Request Input: {}", input_string);
-    let input: KeyProviderInput = serde_json::from_str::<KeyProviderInput>(&input_string)?;
-    let base64_annotation = input
-        .keyunwrapparams
-        .annotation
-        .ok_or_else(|| anyhow!("The annotation field in the input is Empty!"))?;
-    let vec_annotation = base64::decode(base64_annotation)?;
-    let jsonstring_annotation: &str = str::from_utf8(&vec_annotation)?;
-    let dc = input
-        .keyunwrapparams
-        .dc
-        .ok_or_else(|| anyhow!("The Dc field in the input is None!"))?;
-
-    /*
-     * AA expects the received DC parameter format is:
-     * "dc":{
-     *     "Parameters":{
-     *         "attestation_agent":["< KBC_NAME::KBS_URI (base64encode) >"]
-     *     }
-     * }
-     */
-    let parameters_list = dc.parameters.get("attestation-agent").ok_or(anyhow!(
-        "Invalid parameters: the request is not sent to attention agent!"
-    ))?;
-    let kbc_kbs_pair_byte = base64::decode(parameters_list[0].clone())?;
-    let kbc_kbs_pair = std::str::from_utf8(&kbc_kbs_pair_byte)?;
-    if let Some(index) = kbc_kbs_pair.find("::") {
-        let kbc_name: String = kbc_kbs_pair[..index].to_string();
-        let kbs_uri: String = kbc_kbs_pair[(index + 2)..].to_string();
-        debug!("Get KBC_NAME:{}, KBS_URI:{}", kbc_name, kbs_uri);
-        return Ok(InputPayload {
-            kbc_name: kbc_name,
-            kbs_uri: kbs_uri,
-            annotation: jsonstring_annotation.to_string(),
-        });
+impl TryFrom<KeyProviderInput> for InputPayload {
+    type Error = anyhow::Error;
+
+    fn try_from(kpi: KeyProviderInput) -> Result<Self, Self::Error> {
+        let base64_annotation = kpi
+            .keyunwrapparams
+            .annotation
+            .ok_or_else(|| anyhow!(ERR_ANNOTATION_EMPTY))?;
+
+        let vec_annotation = base64::decode(base64_annotation)?;
+
+        let jsonstring_annotation: &str = str::from_utf8(&vec_annotation)?;
```

Comment:
shouldn't we check at this point that the `jsonstring_annotation` is valid JSON?

Reply:
I believe that string should be just plain text at this point. I think that ariable name is erroneous, I've renamed it to `annotation` to avoid the confusion, but it would be good to get confirmation from @jialez0.